### PR TITLE
Ch 380950 button order

### PR DESCRIPTION
**Pull request by conorhanrahan-okta**
## Description:

Removing some of the hacky methods of switching the save/cancel buttons and using the new boolean option from this PR:
https://github.com/okta/okta-ui/pull/3679

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
The before/afters all look the same because this is supposed to be an implementation change only

AdminConsentRequiredController (before)
350521-acrc-before
AdminConsentRequiredController (after)
350521-acrc-after
AdminConsentView (before)
350521-acv-before
AdminConsentView (after)
350521-acv-after
ConsentRequiredController (before)
350521-crc-before
ConsentRequiredController (after)
350521-crc-after
EndUserConsentView (before)
350521-eucv-before
EndUserConsentView (after)
350521-eucv-after

### Issue:

- [OKTA-380950](https://oktainc.atlassian.net/browse/OKTA-380950)

### CODE REVIEW

File: src/AdminConsentRequiredController.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ const ConsentHeader = View.extend({
 
 const ConsentForm = {
   noCancelButton: false,
+  showCancelBeforeSave: true,
```
Comment (pradeepdewda-okta) [inline src/AdminConsentRequiredController.js:62]: added a comment on your courage jira..  I feel this is too specific for this use case and may not be a good candidate for courage change..

**Comment by sherizada-okta** [comment]
Heads up that this will likely miss the monthly release (tomorrow around 1pm), so you will need to rebase this branch off of a different base branch, I think it'll be 5.5? 
SIW releases are Thursday to make the monolith code freeze on Friday.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@okta/okta-signin-widget",
   "description": "The Okta Sign-In Widget",
-  "version": "5.5.0",
+  "version": "5.6.0",
```
Comment (pradeepdewda-okta) [inline package.json:4]: I don't think you need to upgrade this version here..

File: test/unit/helpers/dom/Form.js [hunk 3]
```diff
@@ -65,23 +65,25 @@ export default Dom.extend({
...
     // container holds input and error description
 
-    const errorId = $container.attr('aria-describedby');
+    // const errorId = $container.attr('aria-describedby');
 
-    if (!errorId) {
-      throw new Error('Expected "aria-describedby" attribute for the error container on field: ' + field);
-    }
+    // Uncomment after OKTA-382383
+    // if (!errorId) {
```
Comment (pradeepdewda-okta) [inline test/unit/helpers/dom/Form.js:71]: Do you plan to commit without this test ?

Comment (conorhanrahan-okta) [reply test/unit/helpers/dom/Form.js:71]: Lester Choi is looking into that ticket now. I can merge this with the commented out test, or I can wait until Lester has a fix and cherry-pick his fix to his PR. I'm open to either

Comment (pradeepdewda-okta) [reply test/unit/helpers/dom/Form.js:71]: I suspect if any selenium test fails in okta-core when you try to merge..  If Lester is working actively in plan to merge this week, then i guess you can go ahead if bacon is green for okta-core..

You can kick a downstream build in bacon to see if okta-core is green for this widget change..

File: package.json (hunk 2 above)
Comment (conorhanrahan-okta) [reply package.json:4]: Ah good catch! This was the result of a sloppy rebase

Comment (conorhanrahan-okta) [reply package.json]: I removed this commit

File: test/unit/helpers/dom/Form.js (hunk 3 above)
Comment (conorhanrahan-okta) [reply test/unit/helpers/dom/Form.js:71]: `okta-core` tests have already passed on Bacon, and I've merged the ui and `okta-core` PRs. I also saw all the SIW tests passed on Bacon after cherry-picking Patrick's commit and commenting out this test here.

Comment (pradeepdewda-okta) [reply test/unit/helpers/dom/Form.js:71]: I meant when you merge this change to SIW and then widget gets upgraded to okta-core, then okta-core should pass..

Comment (conorhanrahan-okta) [reply test/unit/helpers/dom/Form.js:71]: Ah, I didn't think about that! Lots of moving parts on this change haha. Ok, I'll make sure that passes before merging this one

Comment (conorhanrahan-okta) [reply test/unit/helpers/dom/Form.js:71]: Looks like it passed: https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-f7e9d9c-60664087

**Comment by haishengwu-okta** [comment]
There is another file could you help to clean up

`okta-signin-widget/src/v2/view-builder/views/consent/EnduserEmailConsentView.js`

**Comment by conorhanrahan-okta** [comment]
> There is another file could you help to clean up
> 
> `okta-signin-widget/src/v2/view-builder/views/consent/EnduserEmailConsentView.js`

That file doesn't seem to be modifying button order itself. It is inheriting from `AdminConsentView` which did have the old button order hack, but that is covered in this PR

**Comment by haishengwu-okta** [comment]
@conorhanrahan-okta 
https://github.com/okta/okta-signin-widget/blob/5.5/src/v2/view-builder/views/consent/EnduserEmailConsentView.js#L55-L63

**Comment by conorhanrahan-okta** [comment]
> @conorhanrahan-okta
> https://github.com/okta/okta-signin-widget/blob/5.5/src/v2/view-builder/views/consent/EnduserEmailConsentView.js#L55-L63

Ah, I was looking at EndUserConsentView, not EndUserEmailConsentView... Ok, I'll clean that up 👍

**Comment by conorhanrahan-okta** [comment]
> @conorhanrahan-okta
> https://github.com/okta/okta-signin-widget/blob/5.5/src/v2/view-builder/views/consent/EnduserEmailConsentView.js#L55-L63

Just pushed a fix

**Comment by conorhanrahan-okta** [comment]
Got another green on Bacon for `okta-core`: https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&branch=d16t-okta-signin-widget-2a91d06-6067689e This is ready for review again

**Review by haishengwu-okta — APPROVED** [review]
Thanks!

**Pull request closed without merging**